src/strings-context.ts

```diff
@@ -0,0 +1,426 @@
+import Crowdin, {
+    ProjectsGroupsModel,
+    SourceStringsModel,
+    StringTranslationsModel,
+    TranslationMemoryModel,
+    GlossariesModel,
+    ScreenshotsModel,
+} from '@crowdin/crowdin-api-client';
+import axios from 'axios';
+import * as path from 'path';
+import ImageAnnotator from './util/image-annotator';
+
+interface AdditionalNodes {
+    otherLanguageTranslations: boolean;
+    glossaryTerms: boolean;
+    tmSuggestions: boolean;
+    screenshots: boolean;
+}
+
+interface StringsContextArgs {
+    client: Crowdin;
+    projectId: number;
+    stringIds: number[];
+    targetLanguagesIds: string[];
+    additionalNodes?: AdditionalNodes;
+    stringsLimit?: number;
+    translationsLimit?: number;
+}
+
+interface Variables {
+    projectId: number;
+    stringIds: number[];
+    stringsLimit: number;
+    languageIds: string[];
+    translationsLimit?: number;
+}
+
+interface ExtendedStringNode extends SourceStringsModel.String {
+    translations: {
+        edges: {
+            node: StringTranslationsModel.StringTranslation;
+        }[];
+    };
+}
+
+interface TranslationMemorySuggestion extends TranslationMemoryModel.ConcordanceSearchResponse {
+    languageId: string;
+}
+
+interface AnnotatedScreenshot extends Partial<ScreenshotsModel.Screenshot> {
+    originalUrl: string;
+}
+
+interface GetStringsContextResult {
+    project: Partial<ProjectsGroupsModel.Project>;
+    strings: ExtendedStringNode[];
+    screenshots?: AnnotatedScreenshot[];
+    tmSuggestions?: TranslationMemorySuggestion[];
+    glossaryTerms?: GlossariesModel.ConcordanceSearchResponse[];
+}
+
+function buildQuery(additionalNodes: AdditionalNodes, getProjectData = true): string {
```

Comment:
maybe give it a bit better name, what exactly query it will prepare for us? Maybe comment/jsdoc